server/models.py

```diff
@@ -112,12 +113,23 @@ class StringList(types.TypeDecorator):
 
     def process_bind_param(self, string_list, dialect):
         # Python -> SQL
-        return ' '.join(string_list)
+        items = []
+        for item in string_list:
+            if " " in item:
+                items.append('"{}"'.format(item))
+            else:
+                items.append(item)
+        return ' '.join(items)
 
     def process_result_value(self, value, dialect):
         # SQL -> Python
-        return value.split()
-
+        values = []
+        for val in shlex.split(value):
```

Comment:
Yeah - I couldn't find a less fragile way to do it (most solutions involved some kind of regex)